api/loader/resource.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2020 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package loader
+
+import (
+	"context"
+	"log"
+	"os"
+
+	"github.com/yujunz/go-getter"
+	"sigs.k8s.io/kustomize/api/filesys"
+	"sigs.k8s.io/kustomize/api/ifc"
+	"sigs.k8s.io/kustomize/api/internal/git"
+)
+
+type resourceSpec struct {
+	// Raw is the original resource in kustomization.yaml
+	Raw string
+
+	// Dir is where the resource is saved
+	Dir filesys.ConfirmedDir
+}
+
+// Getter is a function that can gets resource
+type resourceGetter func(rs *resourceSpec) error
+
+func newResourceGetter(raw string, fSys filesys.FileSystem, referrer *fileLoader, cloner git.Cloner, getter resourceGetter) (ifc.Loader, error) {
+	rs := &resourceSpec{
+		Raw: raw,
+	}
+
+	cleaner := func() error {
+		return fSys.RemoveAll(rs.Dir.String())
+	}
+
+	if err := getter(rs); err != nil {
+		cleaner()
+		return nil, err
+	}
+
+	// TODO(yujunz): check file or directory
+
+	return &fileLoader{
+		loadRestrictor: RestrictionRootOnly,
```

Comment:
The loadRestrictor should be inherited from the parent overlay.

Reply:
This is the top level similar to`newLoaderAtGiClone`  https://github.com/kubernetes-sigs/kustomize/blob/3a2635bd2d7f9d52f251b230df22b7fd5e8bada2/api/loader/fileloader.go#L226

Reply:
Yes, you're correct. For the remote target, we should restrict the load restrictor to be within that remote target.